lib/simplecov/result_merger.rb

```diff
@@ -62,7 +62,7 @@ def results
       def merged_result
         merged = {}
         results.each do |result|
-          merged = result.original_result.merge_resultset(merged)
+          merged = SimpleCov::MergeHelper.merge_resultsets(result.original_result, merged)
```

Comment:
I don't have a strong feeling about it except that this something I've been meaning to do or hoping someone would do for months so 👍 to you!  Only real concern is that we want to make a small public interface so that we can change internal details. So, maybe a method on ResultMerger that calls the helper or whatever, and we can change the whatever however whenever :)